### DISCUSSION TITLE
Encountering issues while DIY GemX

### DISCUSSION THREAD

**Discussion by norulers** [ExpressLRS/ExpressLRS#2959] [General]
I use two E80-900M2213S modules . 
ESP32 S3 MCU.
firmware version V3.5.0 b3b43dd
Add the following code to targets.json
`"tx_dual": {
            "s3": {
                "product_name": "DIY 2.4/900 TX",
                "lua_name": "DIY X-Band",
                "layout_file": "Generic LR1121 Gemini.json",
                "upload_methods": ["uart", "wifi", "etx"],
                "min_version": "3.4.0",
                "platform": "esp32-s3",
                "firmware": "Unified_ESP32S3_LR1121_TX"
            }`
Modify the Generic LR1121 Gemini.json file to match pin definitions.
`{
    "serial_rx": 44,
    "serial_tx": 43,

    "radio_miso": 15,
    "radio_mosi": 16,
    "radio_sck": 7,

    "radio_busy": 4,
    "radio_dio1": 5,
    "radio_nss": 17,
    "radio_rst": 6,

    "radio_busy_2": 18,
    "radio_dio1_2": 8,
    "radio_nss_2": 46,
    "radio_rst_2": 3,

    "power_min": 0,
    "power_high": 3,
    "power_max": 3,
    "power_default": 0,
    "power_control": 0,
    "power_values": [12,16,19,22],
    "power_values_dual": [-10,-6,-3,1],
    "radio_rfsw_ctrl": [7,0,2,3,1,0,4,0],

    "led_rgb": 48,
    "led_rgb_isgrb": true,
    "ledidx_rgb_status": [0],
    "ledidx_rgb_boot": [0],

    "button": 0,

    "radio_dcdc": true,
    
    "use_backpack": true,
    "debug_backpack_baud": 460800,
    "debug_backpack_rx": 14,
    "debug_backpack_tx": 13,
    "backpack_boot": 21,
    "backpack_en": 38,
    "passthrough_baud": 230400
}`
LUA freezes when connecting module to TX.

https://youtu.be/nt4NAhVlKPU

**Comment by norulers** [comment]
I upgraded the LR1121 firmware on WIFI, and the firmware version was upgraded from 101 to 103. However, the LUA freeze did not improve.
![20240928192624](https://github.com/user-attachments/assets/60e9f8b8-077c-4ebb-a5a6-0aaa46db8d01)

**Reply by NanoNM** [reply]
where i can git lr1121 firmware

**Reply by pkendall64** [reply]
The firmware files are here https://github.com/Lora-net/radio_firmware_images/tree/master/lr1121/transceiver
You should upload the .bin file

**Reply by NanoNM** [reply]
when i upload the 101 or 103bin file，1121.html recognized 0x03, but after powering on again, lr1121 cannot recognize it

**Reply by NanoNM** [reply]
ExpressLRS TX Module Booted...
Config version 7
Hal Init
LR1121 Reset
version is
SX12XX_Radio_1 1
Read LR1121 #1 Use Case (0x03 = LR1121): 223
LR1121 #1 failed to be detected.
Begin Webupdater
Changing to AP mode
HTTPUpdateServer ready! Open http://elrs_tx.local in your browser

**Reply by NanoNM** [reply]
i fix my problem but new problem happend i can not band a 2.4g rx

**Reply by TURALLATURAL** [reply]
Hi can you bind , i have same issue i solve freeze problems, but i cnat bind to 915

**Comment by pkendall64** [comment]
Everything looks good in the JSON files, but without a schematic it's hard to tell if it's wired correctly.

**Reply by norulers** [reply]
Here is the schematic
[DIY GemX.pdf](https://github.com/user-attachments/files/17177058/DIY.GemX.pdf)

**Reply by pkendall64** [reply]
For a start you need ~220 ohm buffer resistors on the MISO lines to the MCU.

**Reply by norulers** [reply]
Thank you for your reply. I will test it

**Comment by norulers** [comment]
![20241006143508](https://github.com/user-attachments/assets/1f80f1ea-61be-4bae-a8d0-2ac937e06671)
![20241006143529](https://github.com/user-attachments/assets/1af37ffd-1bc5-4726-8206-b3b22bfa9538)
![20241006143543](https://github.com/user-attachments/assets/f598dc9b-bfb5-4058-a172-ab751774414e)
![20241006143559](https://github.com/user-attachments/assets/35da1c5e-9b2c-45ad-bb3f-a3fe6a094a65)
The problem persists even after adding a 220 ohm resistor.

**Comment by norulers** [comment]
![20241008091408](https://github.com/user-attachments/assets/d1e5cded-ce43-4e22-8962-09e18f37c693)
The TCXO crystal oscillator power supply of E80 module is provided by LR1121 VTCXO. Is it necessary to set the output power supply to make it work ？
![20241008092154](https://github.com/user-attachments/assets/b498314d-b6ec-4cd1-a762-b82bbfff0174)

**Comment by norulers** [comment]
![20241008095625](https://github.com/user-attachments/assets/1b520de8-c5e0-4394-ae08-65a1f471d978)
The problem disappeared after using external 3.3V to power TCXO.

**Reply by NanoNM** [reply]
can u band u rx？ i can not band 2.4g rx.

i am not try on 915rx yet

**Comment by pkendall64** [comment]
Yes it would be necessary to make code changes to select the TCXO voltage. We currently don't do that in the firmware. Should be an easy thing to add, but all the current modules and manufactured devices directly power the TCXO

**Reply by norulers** [reply]
It would be great if you could add code to adjust the voltage.
Open the demo program of the E80 module and find that the TCXO voltage is set to 1.8V
![20241008164640](https://github.com/user-attachments/assets/363eb441-b5f3-4292-a012-62e821b6442b)

**Reply by norulers** [reply]
I see that the lr1121-SET_TCXO-MODE branch already contains code. I will test it later.
https://github.com/ExpressLRS/ExpressLRS/tree/lr1121-SET_TCXO_MODE

**Reply by NanoNM** [reply]
is worked？
![image](https://github.com/user-attachments/assets/6063cd1a-0aa9-4c6b-81fc-818f90d48c71)

**Comment by aelmod** [comment]
Did you manage to fix the problem that TX freezes when trying to change something in Lua?

Upd: fixed by setting correct TCXO

**Reply by TURALLATURAL** [reply]
Hi can you help me please, i had problems with freeze when i binding , but now bind is ok but it cant connsct to rx

**Comment by StreakingJerry** [comment]
Hi, is this fixing merged into the main branch? I tried to search the TCXO in main branch but don't find related code
